pkg/cas/content_addressable_storage_server.go

```diff
@@ -47,7 +47,36 @@ func (s *contentAddressableStorageServer) FindMissingBlobs(ctx context.Context,
 }
 
 func (s *contentAddressableStorageServer) BatchReadBlobs(ctx context.Context, in *remoteexecution.BatchReadBlobsRequest) (*remoteexecution.BatchReadBlobsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "This service does not support batched reading of blobs")
+	// Asynchronously call Get() for every blob.
+	responsesChan := make(chan *remoteexecution.BatchReadBlobsResponse_Response, len(in.Digests))
+	for _, reqDigest := range in.Digests {
+		go func(reqDigest *remoteexecution.Digest) {
```

Comment:
This places no upper bound on the number of concurrent RPCs this would generate. If a caller passes in a list of thousands of digests, this would spawn thousands of goroutines, which could cause thousands of network connections/channels/.... to be opened to storage backends, etc.

For an initial implementation, I would suggest that no concurrency is added. If it turns out that this kills performance, we can always add it later on either by adding a constant limit on concurrency or by pushing support for batch reading/writing into BlobAccess itself. The latter would prevent batched calls from ending up being decomposed in case bb-storage instances are chained/stacked.

I would also suggest that BatchUpdateBlobs() is adjusted accordingly; it seemingly also has no such limit.